workflow/artifacts/artifacts.go

```diff
@@ -79,7 +78,23 @@ func NewDriver(ctx context.Context, art *wfv1.Artifact, ri resource.Interface) (
 		return &driver, nil
 	}
 	if art.HTTP != nil {
-		return &http.ArtifactDriver{}, nil
+		if art.HTTP.UsernameSecret != nil && art.HTTP.PasswordSecret != nil {
```

Comment:
I think this needs to be the same as below, you can have a password without a users (e.g. when you use a Personal Access Token)